packages/ethereum/contracts/src/node-stake/NodeStakeFactory.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "openzeppelin-contracts-upgradeable/proxy/ClonesUpgradeable.sol";
+import "openzeppelin-contracts/utils/Address.sol";
+import "../../script/utils/SafeSuiteLib.sol";
+import "safe-contracts/proxies/SafeProxy.sol";
+import "safe-contracts/proxies/SafeProxyFactory.sol";
+import "safe-contracts/Safe.sol";
+import "safe-contracts/common/Enum.sol";
+
+
+contract HoprNodeStakeFactory  {
+    using Address for address;
+    using ClonesUpgradeable for address;
+
+    address internal constant SENTINEL_OWNERS = address(0x1);
+    bytes32 internal r;
```

Comment:
why should this only be internal instead of public? wouldn't this be good to be immutable?